test/ngAnimate/animateCssDriverSpec.js

```diff
@@ -129,8 +129,14 @@ describe("ngAnimate $$animateCssDriver", function() {
           $rootElement.append(from);
           $rootElement.append(to);
 
-          // we need to do this so that style detection works
-          $$body.append($rootElement);
+          var doc = $document[0];
+
+          // there is one test in here that expects the rootElement
+          // to superceed the body node
+          if (!$rootElement[0].contains(doc.body)) {
+            // we need to do this so that style detection works
+            jqLite(doc.body).append($rootElement);
+          }
```

Comment:
Doing this kind of conditional work in `beforeEach` makes me feel uncomfortable.
Actually almost all `beforeEach` blocks in general make me uncomfortable, especially when the spec file has over a thousand lines.
I often find myself looking at a spec that clearly has some magic happening somewhere in the file but there are potentially multiple nested decribe blocks who may or may not provide a `beforeEach`.
As a general rule I am moving more toward a style of helpers, except for very basic setup like getting hold of injectables.

But anyway, not to worry here on this occasion.